SteamTrade/Inventory/Inventory.cs

```diff
@@ -0,0 +1,245 @@
+        public InventoryType InventoryType { get; private set; }
+        public IEnumerable<InventoryItem> Items { get; private set; }
+        public bool InventoryLoaded { get; private set; }
+
+        private SteamWeb web;
+        private FetchType fType;
+        private ulong start;
+        private bool callbackFired;
+
+        public Inventory(SteamWeb web, SteamID owner, InventoryType type, FetchType fType = FetchType.Inventory, ulong iStart = 0)
+        {
+            this.web = web;
+            InventoryOwner = owner;
+            InventoryType = type;
+            this.fType = fType;
+            start = iStart;
+        }
+
+        public static Inventory FetchInventory(SteamWeb web, SteamID owner, InventoryType type, FetchType fType = FetchType.Inventory)
+        {
+            Inventory inv = new Inventory(web, owner, type, fType);
+            inv.FetchInventory();
+            return inv;
+        }
+
+        public static void FetchInventories(IEnumerable<Inventory> inventories)
+        {
+            foreach (Inventory inv in inventories)
+                inv.FetchInventory();
+        }
+
+        public static IEnumerable<Inventory> FetchInventories(SteamWeb web, SteamID owner, IEnumerable<InventoryType> invTypes, FetchType fType = FetchType.Inventory)
+        {
+            List<Inventory> invs = new List<Inventory>();
+            foreach (InventoryType type in invTypes)
+                invs.Add(Inventory.FetchInventory(web, owner, type, fType));
+            return invs;
+        }
+
+#pragma warning disable 4014
+        public static void QuickFetchInventory(Inventory inventory, OnInventoryLoaded callback)
+        {
+            inventory.FetchInventoryAsync(callback);
+        }
+
+        public static void QuickFetchInventory(SteamWeb web, SteamID owner, OnInventoryLoaded callback, InventoryType type, FetchType fType = FetchType.Inventory)
+        {
+            new Inventory(web, owner, type, fType).FetchInventoryAsync(callback);
+        }
+
+        public static void QuickFetchInventories(IEnumerable<Inventory> inventories, OnInventoryLoaded callback)
+        {
+            foreach (Inventory inv in inventories)
+                inv.FetchInventoryAsync(callback);
+        }
+
+        public static void QuickFetchInventories(SteamWeb web, SteamID owner, OnInventoryLoaded callback, IEnumerable<InventoryType> invTypes, FetchType fType = FetchType.Inventory)
+        {
+            foreach (InventoryType type in invTypes)
+                new Inventory(web, owner, type, fType).FetchInventoryAsync(callback);
+        }
+#pragma warning restore 4014
+
+        public void FetchInventory()
+        {
+            Parse();
+        }
+
+        public async Task FetchInventoryAsync(OnInventoryLoaded callback)
+        {
+            await ParseAsync(callback);
+        }
+
+#warning "This non-async method needs debugging as something is horrible with it."
```

Comment:
@BlueRaja this is here until I can fix the Parse() method... The ParseAsync() method works perfectly and Parse() is a copy of it.